nupic/research/frameworks/vernon/experiments/supervised_experiment.py

```diff
@@ -79,6 +79,7 @@ def __init__(self):
         self.total_batches = 0
         self.epochs_to_validate = []
         self.current_epoch = 0
+        self.rank = 0
```

Comment:
The `rank` field is only available if the experiment is extending [DistributedBase](https://github.com/numenta/nupic.research/blob/4909fe6b79abc292b3009c0305aa56f84327b395/nupic/research/frameworks/vernon/distributed/experiments/components/distributed_base.py#L61). It should not be used in `ExperimentBase`. I think it would be cleaner to remove `rank` from the logging in the `post_batch` method